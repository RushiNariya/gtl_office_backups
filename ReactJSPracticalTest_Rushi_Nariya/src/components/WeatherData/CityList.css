.city-list {
  list-style-type: none;
  margin: 20px 20px 0;
  padding: 0;
  color: #fff;
}

.city-list .city-element {
  margin: 0;
  padding: 0;
}
