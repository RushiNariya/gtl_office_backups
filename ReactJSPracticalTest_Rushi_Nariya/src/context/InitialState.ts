/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

import { createContext } from "react";
import cityInfo from "../intefaces/cityInfo";
import weatherInfo from "../intefaces/weatherInfo";

export type initState = {
  data: weatherInfo;
  city: Array<cityInfo>;
  error: string;
  searchByCityHandler: (data: weatherInfo) => void;
  searchCityHandler: (data: cityInfo) => void;
  setWeatherDataHandler: (data: weatherInfo) => void;
  errorHandler: (error: string) => void;
  resetSetErrorHandler: () => void;
  removeCityHandler: (city: string) => void;
};
  
const initialState: initState = {
  data: {},
  city: [],
  error: '',
  searchByCityHandler: (data) => {},
  searchCityHandler: (data) => {},
  setWeatherDataHandler: (data) => {},
  errorHandler: (error) => {},
  resetSetErrorHandler: () => {},
  removeCityHandler: (city) => {},
};

export const GlobalContext = createContext(initialState);

export default initialState;