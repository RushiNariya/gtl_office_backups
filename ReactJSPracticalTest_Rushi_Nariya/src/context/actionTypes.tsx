export const SEARCH_BY_CITY = 'SEARCH_BY_CITY';
export const SET_WEATHER_DATA = 'SET_WEATHER_DATA';
export const SEARCH_CITY = 'SEARCH_CITY';
export const REMOVE_CITY = 'REMOVE_CITY';
export const SET_ERROR = 'SET_ERROR';
export const RESET_ERROR = 'RESET_ERROR';
export const REFRESH_STATE = 'REFRESH_STATE';