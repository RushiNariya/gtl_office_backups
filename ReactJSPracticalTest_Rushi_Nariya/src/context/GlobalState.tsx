/* eslint-disable @typescript-eslint/no-non-null-assertion */
/* eslint-disable @typescript-eslint/no-empty-function */
import React, { useReducer, useEffect } from 'react';
import PropTypes from 'prop-types';
import AppReducer from './reducer';
import {
  searchByCityAction,
  searchCityAction,
  setWeatherDataAction,
  setErrorAction,
  resetErrorAction,
  removeCityAction,
  refreshStateAction,
} from './actions';
import weatherInfo from '../intefaces/weatherInfo';
import cityInfo from '../intefaces/cityInfo';
import { GlobalContext, initState } from './InitialState';


type props = {
  children: any;
  initialState: initState;
};

export const GlobalProvider = ({ children, initialState }: props): JSX.Element => {
  const [state, dispatch] = useReducer(AppReducer, initialState);

  function searchByCityHandler(data: weatherInfo): void {
    dispatch(searchByCityAction(data));
  }
  function searchCityHandler(city: cityInfo): void {
    dispatch(searchCityAction(city));
  }
  function removeCityHandler(city: string): void {
    dispatch(removeCityAction(city));
  }
  function setWeatherDataHandler(data: weatherInfo): void {
    dispatch(setWeatherDataAction(data));
  }
  function errorHandler(error: string): void {
    dispatch(setErrorAction(error));
  }
  function resetSetErrorHandler(): void {
    dispatch(resetErrorAction());
  }

  function refreshState(): void {
    const refreshedState: Array<cityInfo> = localStorage.getItem('state')
      ? JSON.parse(localStorage.getItem('state')!)
      : initialState.city;
    dispatch(refreshStateAction(refreshedState));
  }

  useEffect(() => {
    refreshState();
  }, []);

  useEffect(() => {
    localStorage.setItem('state', JSON.stringify(state.city));
  }, [state]);

  return (
    <GlobalContext.Provider
      value={{
        data: state.data,
        city: state.city,
        error: state.error,
        searchByCityHandler,
        searchCityHandler,
        setWeatherDataHandler,
        errorHandler,
        resetSetErrorHandler,
        removeCityHandler,
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};

GlobalProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
