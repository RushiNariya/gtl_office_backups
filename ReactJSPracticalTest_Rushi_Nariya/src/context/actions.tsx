import cityInfo from '../intefaces/cityInfo';
import weatherInfo from '../intefaces/weatherInfo';
import {
  REFRESH_STATE,
  REMOVE_CITY,
  RESET_ERROR,
  SEARCH_BY_CITY,
  SEARCH_CITY,
  SET_ERROR,
  SET_WEATHER_DATA,
} from './actionTypes';
import { initState } from './InitialState';

export interface ActionMethodType {
  type: string;
  payload: weatherInfo | string | initState | Array<cityInfo>;
}

export const searchByCityAction = (data: weatherInfo): ActionMethodType => ({
  type: SEARCH_BY_CITY,
  payload: data,
});

export const setWeatherDataAction = (data: weatherInfo): ActionMethodType => ({
  type: SET_WEATHER_DATA,
  payload: data,
});

export const searchCityAction = (data: cityInfo): ActionMethodType => ({
  type: SEARCH_CITY,
  payload: data,
});

export const removeCityAction = (city: string): ActionMethodType => ({
  type: REMOVE_CITY,
  payload: city,
});

export const setErrorAction = (error: string): ActionMethodType => ({
  type: SET_ERROR,
  payload: error,
});

export const resetErrorAction = (): ActionMethodType => ({
  type: RESET_ERROR,
  payload: '',
});

export const refreshStateAction = (
  state: Array<cityInfo>
): ActionMethodType => ({
  type: REFRESH_STATE,
  payload: state,
});
