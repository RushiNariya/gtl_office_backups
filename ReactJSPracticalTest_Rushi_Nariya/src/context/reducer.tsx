/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import {
  SEARCH_BY_CITY,
  SEARCH_CITY,
  SET_WEATHER_DATA,
  SET_ERROR,
  RESET_ERROR,
  REMOVE_CITY,
  REFRESH_STATE,
} from './actionTypes';
import { initState } from './InitialState';

const AppReducer = (state: initState, action: any): initState => {
  switch (action.type) {
    case SEARCH_BY_CITY:
      return {
        ...state,
      };
    case SET_WEATHER_DATA:
      return {
        ...state,
        data: action.payload,
      };
    case SEARCH_CITY:
      if (state.city.find((item) => item.name === action.payload.name)) {
        return state;
      }
      return {
        ...state,
        city: [...state.city, action.payload],
      };
    case REMOVE_CITY:
      return {
        ...state,
        data: {},
        city: state.city.filter((item) => item.name !== action.payload),
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    case RESET_ERROR:
      return {
        ...state,
        error: '',
      };
    case REFRESH_STATE:
      return {
        ...state,
        city: action.payload,
        error: '',
      };
    default:
      return state;
  }
};

export default AppReducer;
