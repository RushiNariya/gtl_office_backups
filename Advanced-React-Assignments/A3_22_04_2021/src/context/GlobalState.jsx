import React, { createContext, useReducer } from 'react';
import PropTypes from 'prop-types';
import reducer, { initialState } from './reducer';
import {
  playAction,
  pauseAction,
  playNextAction,
  playPreviousAction,
  jumpToSlideAction,
} from './ActionMethods';

export const GlobalContext = createContext(initialState);

export const GlobalProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  function playImage() {
    dispatch(playAction());
  }
  function pauseImage() {
    dispatch(pauseAction());
  }
  function playNextImage(id) {
    dispatch(playNextAction(id));
  }
  function playPreviousImage(id) {
    dispatch(playPreviousAction(id));
  }
  function jumpToSlideImage(id) {
    dispatch(jumpToSlideAction(id));
  }

  return (
    <GlobalContext.Provider
      value={{
        state,
        playImage,
        pauseImage,
        playNextImage,
        playPreviousImage,
        jumpToSlideImage,
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};

GlobalProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
