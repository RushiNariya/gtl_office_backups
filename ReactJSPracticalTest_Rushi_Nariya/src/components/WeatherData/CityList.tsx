import React, { useEffect, useContext } from 'react';
import toast from 'react-hot-toast';
import { getCurrentWeatherDataApi, getPosition } from '../../Api/weatherAPI';
import City from './CityData/City';
import './CityList.css';
import cityInfo from '../../intefaces/cityInfo';
import { GlobalContext } from '../../context/InitialState';

function WeatherData(): JSX.Element {
  const { searchCityHandler, errorHandler, resetSetErrorHandler, city } =
    useContext(GlobalContext);

  const getWeather = async (latitude: any, longitude: any): Promise<any> => {
    const data = await getCurrentWeatherDataApi(latitude, longitude);
    if (data.cod === 200) {
      resetSetErrorHandler();

      const cityData: cityInfo = {
        name: data.name,
        country: data.sys.country,
        temp: data.main.temp,
        icon: data.weather[0].icon,
        longitude: data.coord.lon,
        latitude: data.coord.lat,
        maxTemp: data.main.temp_max,
        minTemp: data.main.temp_min,
      };
      searchCityHandler(cityData);
    } else {
      errorHandler(data.message);
    }
  };

  useEffect(() => {
    getPosition()
      .then((position: any) => {
        getWeather(position.coords.latitude, position.coords.longitude);
        setInterval(
          () => getWeather(position.coords.latitude, position.coords.longitude),
          3600000
        );
      })
      .catch((err) => {
        toast.error(err.message);
      });
  }, []);

  return (
    <>
      {city.length ? (
        <ul className="city-list">
          {city.map((item) => {
            return (
                <li key={item.name} className="city-element">
                  <City cityData={item} />
                </li>
            );
          })}
        </ul>
      ) : <div className="city-list">No cities Found!</div>}
    </>
  );
}

export default WeatherData;
