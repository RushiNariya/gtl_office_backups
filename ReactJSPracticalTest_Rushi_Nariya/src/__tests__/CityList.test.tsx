/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import CityList from '../components/WeatherData/CityList';
import { GlobalProvider } from '../context/GlobalState';
import initialState from '../context/InitialState';

let wrapped: ReactWrapper;

initialState.city = [
  {
    name: 'Sarkhej',
    temp: 31.04,
    icon: '50d',
    latitude: 23.0439,
    longitude: 72.5086,
    minTemp: 31.04,
    maxTemp: 31.04,
    country: 'IN',
  },
  {
    name: 'Pune',
    temp: 32.04,
    icon: '50d',
    latitude: 21.0439,
    longitude: 74.5086,
    minTemp: 33.04,
    maxTemp: 31.04,
    country: 'IN',
  },
];

describe('Test Case For CityList Component', (): void => {
  beforeEach((): void => {
    wrapped = mount(
      <GlobalProvider initialState={initialState}>
        <CityList />
      </GlobalProvider>
    );
  });

  afterEach((): void => {
    wrapped.unmount();
  });

  it('renders the CityList component', (): void => {
    expect(wrapped).toMatchSnapshot();
  });

  it('check city list data length', (): void => {
    expect(wrapped.find('li').length).toEqual(2);
  });

  it('check correct data render on city list', (): void => {
    expect(wrapped.render().text()).toContain('Sarkhej');
    expect(wrapped.render().text()).toContain('Pune');
  });
});
