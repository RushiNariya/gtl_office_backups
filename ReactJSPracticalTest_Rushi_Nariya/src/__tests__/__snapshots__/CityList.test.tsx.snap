// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For AppTitle renders the CityList component 1`] = `
<GlobalProvider
  initialState={
    Object {
      "city": Array [
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 23.0439,
          "longitude": 72.5086,
          "maxTemp": 31.04,
          "minTemp": 31.04,
          "name": "Sarkhej",
          "temp": 31.04,
        },
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 21.0439,
          "longitude": 74.5086,
          "maxTemp": 31.04,
          "minTemp": 33.04,
          "name": "Pune",
          "temp": 32.04,
        },
      ],
      "data": Object {},
      "error": "",
      "errorHandler": [Function],
      "removeCityHandler": [Function],
      "resetSetErrorHandler": [Function],
      "searchByCityHandler": [Function],
      "searchCityHandler": [Function],
      "setWeatherDataHandler": [Function],
    }
  }
>
  <WeatherData>
    <ul
      className="city-list"
    >
      <li
        className="city-element"
        key="Sarkhej"
      >
        <City
          cityData={
            Object {
              "country": "IN",
              "icon": "50d",
              "latitude": 23.0439,
              "longitude": 72.5086,
              "maxTemp": 31.04,
              "minTemp": 31.04,
              "name": "Sarkhej",
              "temp": 31.04,
            }
          }
        >
          <article
            className="city"
            onClick={[Function]}
          >
            <div
              className="column-1"
            >
              <div
                className="name"
              >
                Sarkhej
                <img
                  className="flag"
                  src="http://openweathermap.org/img/wn/50d@2x.png"
                />
              </div>
            </div>
            <div
              className="column-2"
            >
              <div
                className="temp"
              >
                <span
                  className="units"
                >
                  31.04
                </span>
                <span
                  className="metrics"
                >
                  째C
                </span>
              </div>
            </div>
            <button
              className="cancel-button"
              onClick={[Function]}
            >
              <p
                className="icon-cancel"
              >
                x
              </p>
            </button>
          </article>
        </City>
      </li>
      <li
        className="city-element"
        key="Pune"
      >
        <City
          cityData={
            Object {
              "country": "IN",
              "icon": "50d",
              "latitude": 21.0439,
              "longitude": 74.5086,
              "maxTemp": 31.04,
              "minTemp": 33.04,
              "name": "Pune",
              "temp": 32.04,
            }
          }
        >
          <article
            className="city"
            onClick={[Function]}
          >
            <div
              className="column-1"
            >
              <div
                className="name"
              >
                Pune
                <img
                  className="flag"
                  src="http://openweathermap.org/img/wn/50d@2x.png"
                />
              </div>
            </div>
            <div
              className="column-2"
            >
              <div
                className="temp"
              >
                <span
                  className="units"
                >
                  32.04
                </span>
                <span
                  className="metrics"
                >
                  째C
                </span>
              </div>
            </div>
            <button
              className="cancel-button"
              onClick={[Function]}
            >
              <p
                className="icon-cancel"
              >
                x
              </p>
            </button>
          </article>
        </City>
      </li>
    </ul>
  </WeatherData>
</GlobalProvider>
`;

exports[`Test Case For CityList Component renders the CityList component 1`] = `
<GlobalProvider
  initialState={
    Object {
      "city": Array [
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 23.0439,
          "longitude": 72.5086,
          "maxTemp": 31.04,
          "minTemp": 31.04,
          "name": "Sarkhej",
          "temp": 31.04,
        },
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 21.0439,
          "longitude": 74.5086,
          "maxTemp": 31.04,
          "minTemp": 33.04,
          "name": "Pune",
          "temp": 32.04,
        },
      ],
      "data": Object {},
      "error": "",
      "errorHandler": [Function],
      "removeCityHandler": [Function],
      "resetSetErrorHandler": [Function],
      "searchByCityHandler": [Function],
      "searchCityHandler": [Function],
      "setWeatherDataHandler": [Function],
    }
  }
>
  <WeatherData>
    <ul
      className="city-list"
    >
      <li
        className="city-element"
        key="Sarkhej"
      >
        <City
          cityData={
            Object {
              "country": "IN",
              "icon": "50d",
              "latitude": 23.0439,
              "longitude": 72.5086,
              "maxTemp": 31.04,
              "minTemp": 31.04,
              "name": "Sarkhej",
              "temp": 31.04,
            }
          }
        >
          <article
            className="city"
            onClick={[Function]}
          >
            <div
              className="column-1"
            >
              <div
                className="name"
              >
                Sarkhej
                <img
                  className="flag"
                  src="http://openweathermap.org/img/wn/50d@2x.png"
                />
              </div>
            </div>
            <div
              className="column-2"
            >
              <div
                className="temp"
              >
                <span
                  className="units"
                >
                  31.04
                </span>
                <span
                  className="metrics"
                >
                  째C
                </span>
              </div>
            </div>
            <button
              className="cancel-button"
              onClick={[Function]}
            >
              <p
                className="icon-cancel"
              >
                x
              </p>
            </button>
          </article>
        </City>
      </li>
      <li
        className="city-element"
        key="Pune"
      >
        <City
          cityData={
            Object {
              "country": "IN",
              "icon": "50d",
              "latitude": 21.0439,
              "longitude": 74.5086,
              "maxTemp": 31.04,
              "minTemp": 33.04,
              "name": "Pune",
              "temp": 32.04,
            }
          }
        >
          <article
            className="city"
            onClick={[Function]}
          >
            <div
              className="column-1"
            >
              <div
                className="name"
              >
                Pune
                <img
                  className="flag"
                  src="http://openweathermap.org/img/wn/50d@2x.png"
                />
              </div>
            </div>
            <div
              className="column-2"
            >
              <div
                className="temp"
              >
                <span
                  className="units"
                >
                  32.04
                </span>
                <span
                  className="metrics"
                >
                  째C
                </span>
              </div>
            </div>
            <button
              className="cancel-button"
              onClick={[Function]}
            >
              <p
                className="icon-cancel"
              >
                x
              </p>
            </button>
          </article>
        </City>
      </li>
    </ul>
  </WeatherData>
</GlobalProvider>
`;
